### PR TITLE
Fixes #31177 - make registration variables more readable

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: app/controllers/concerns/foreman/controller/registration.rb [hunk 1]
```diff
@@ -19,18 +19,20 @@ def global_registration_vars
...
+    context = {
+      user: User.current,
+      auth_token: User.current.jwt_token!(expiration: 4.hours.to_i),
+      organization: (organization || User.current.default_organization || User.current.my_organizations.first),
+      location: (location || User.current.default_location || User.current.my_locations.first),
+      hostgroup: host_group,
+      operatingsystem: operatingsystem,
+      url_host: registration_url.host,
+      registration_url: registration_url,
+    }
```
Comment (ekohl) [inline app/controllers/concerns/foreman/controller/registration.rb:34]: Perhaps also an empty line here
```suggestion
    }

```

**Review by ekohl — DISMISSED** [review]

File: app/controllers/concerns/foreman/controller/registration.rb [hunk 2]
```diff
@@ -19,18 +19,20 @@ def global_registration_vars
     host_group = Hostgroup.authorized(:view_hostgroups).find(params['hostgroup_id']) if params["hostgroup_id"].present?
     operatingsystem = Operatingsystem.authorized(:view_operatingsystems).find(params['operatingsystem_id']) if params["operatingsystem_id"].present?
 
+    context = {
+      user: User.current,
+      auth_token: User.current.jwt_token!(expiration: 4.hours.to_i),
```
Comment (rabajaj0509) [inline app/controllers/concerns/foreman/controller/registration.rb:25]: Any idea why we are using 4.hours.to_i, does 4 hours has a significance or it is a random time frame?

Comment (ekohl) [reply app/controllers/concerns/foreman/controller/registration.rb:25]: It was there before, just moving things around. However, I would agree that it's ugly that it's hardcoded and not even a constant.

Comment (rabajaj0509) [reply app/controllers/concerns/foreman/controller/registration.rb:25]: makes sense :+1:

Comment (ares) [reply app/controllers/concerns/foreman/controller/registration.rb:25]: a bit of offtopic to this PR: We could extract that to be configurable, however is that necessary? Personally I'd wait until someone asks for it, the first token is configurable already. We have enough settings already. The other option would be to base on the expiration set for the first request. However do not that these two expirations are not the same thing. If you think it's valuable, please describe how we should set this and link the issue to https://projects.theforeman.org/issues/30440.

**Comment by ares** [comment]
PR rebased and empty line added, this should be good for merge

**Review by ekohl — APPROVED** [review]
Thanks!

**Pull request merged**